### PR TITLE
Implement context menu and extra info on double-click for DIP3 masternode list

### PR DESCRIPTION
**Pull request by codablock**
Menu items:
- "Copy ProTx hash"
- "Copy Collateral hash"

On double click:
Show simple message box with the json representation of the DMN.

Example message box (I know it's super ugly...):
image

### CODE REVIEW

File: src/qt/masternodelist.h [hunk 1]
```diff
@@ -85,5 +89,10 @@ private Q_SLOTS:
     void on_startMissingButton_clicked();
     void on_tableWidgetMyMasternodes_itemSelectionChanged();
     void on_UpdateButton_clicked();
+
+    CDeterministicMNCPtr GetSelectedDIP3MN();
```
Comment (UdjinM6) [inline src/qt/masternodelist.h]: This is not a slot, it's just a regular private function, should be in previous section (e.g. line 80).

File: src/qt/masternodelist.cpp [hunk 2]
```diff
@@ -646,3 +667,63 @@ void MasternodeList::ShowQRCode(std::string strAlias)
...
+}
+
+void MasternodeList::on_copyCollateralTxHash_clicked()
+{
+    auto dmn = GetSelectedDIP3MN();
+    if (!dmn) {
+        return;
+    }
+
+    QApplication::clipboard()->setText(QString::fromStdString(dmn->collateralOutpoint.ToStringShort()));
```
Comment (UdjinM6) [inline src/qt/masternodelist.cpp]: Technically speaking, this is not a hash but an outpoint i.e. menu text/function name is misleading.

File: src/qt/masternodelist.cpp [hunk 3]
```diff
@@ -88,8 +88,8 @@ MasternodeList::MasternodeList(const PlatformStyle* platformStyle, QWidget* pare
     connect(ui->tableWidgetMyMasternodes, SIGNAL(doubleClicked(QModelIndex)), this, SLOT(on_QRButton_clicked()));
     connect(startAliasAction, SIGNAL(triggered()), this, SLOT(on_startButton_clicked()));
 
-    QAction* copyProTxHashAction = new QAction(tr("Copy ProTx hash"), this);
-    QAction* copyCollateralTxHashAction = new QAction(tr("Copy Collateral hash"), this);
+    QAction* copyProTxHashAction = new QAction(tr("Copy ProTx Hash"), this);
+    QAction* copyCollateralTxHashAction = new QAction(tr("Copy Collateral Outpoint"), this);
```
[the remark this answers is not in the document]
Comment (codablock) [reply src/qt/masternodelist.cpp]: fixed, sorry about that one :)

**Comment by PastaPastaPasta** [comment]
Once DIP 3 is activated, will this conflict with the current menu that pops up when double clicking on a MN?

**Comment by codablock** [comment]
@PastaPastaPasta Nope, these are two different context menus

**Pull request merged**